javascript/javascript1/week3/NOnoN0nOYes.js

```diff
@@ -0,0 +1,52 @@
+// Save a note
+let note = [];
+function saveNote(content, id) {
+  const addNote = { content: content, id: id };
+  return note.push(addNote);
+}
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+console.log(note);
+
+// get note
+function getNote(id) {
+  for (let i = 0; i < note.length; i++) {
+    if (note[i].id === id) {
+      return note[i];
+    } else if (typeof note[i].id !== Number) {
+      return `Error!! This is not a valid id`;
+    } else {
+      return `Error!! This is not a valid id`;
+    }
+  }
+}
+console.log(getNote(2));
+
+//Log Out Notes
+function logOutNotesFormatted() {
+  for (let i = 0; i < note.length; i++) {
+    const newNote = note[i];
+    console.log(
+      `The note with id: ${newNote.id}, has the following note text: ${newNote.content}`
+    );
+  }
+}
+logOutNotesFormatted();
+
+// Edit note
+
+function editNote(id) {
+  let noteFound = false;
+  for (let i = 0; i < note.length; i++) {
+    if (note[i].id === id) {
+      note[i].content = prompt("Edit your note");
+      console.log(note);
+      noteFound = true;
+      break;
+    }
+  }
+  if (!noteFound) {
+    console.log(`Please enter a valid id`);
+  }
+}
```

Comment:
Great extra feature idea!

If you want to grab the data from the prompt response then try looking into it here:
https://www.w3schools.com/jsref/met_win_prompt.asp

Also, when the note has been found and made it past your if statement then you should make use of `return`..

```
// Pseudo code:
if (note.id === id) {
   ...code to get prompt input value
   return (to finish the function)
}
```

Reply:
thanks for suggestion and edited as per that.